docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md

```diff
@@ -105,27 +102,31 @@ props.setProperty(ReadWriteSplittingPlugin.READER_HOST_SELECTOR_STRATEGY.name, "
-
-There are many session state attributes that can change during a session, and many ways to change them. Consequently, the read/write splitting plugin has limited support for transferring session state between connections. The following attributes will be automatically transferred when switching connections:
+### Session state
 
-- autocommit value
-- transaction isolation level
-
-All other session state attributes will be lost when switching connections between the writer/reader.
-
-If your SQL workflow depends on session state attributes that are not mentioned above, you will need to re-configure those attributes each time that you switch between the writer/reader.
+The plugin supports session state transfer when switching connection. All attributes mentioned in [Session State](../SessionState.md) are automatically transferred to a new connection. 
 
 
 ### Limitations when using Spring Boot/Framework
 
 #### @Transactional(readOnly = True)
 
 > [!WARNING]\
-> The use of read/write splitting with the annotation @Transactional(readOnly = True) is not recommended.
+> The use of read/write splitting with the annotation @Transactional(readOnly = True) is **not** recommended for any configurations except a configuration with internal connection pool. Other configurations will cause a significant performance degradation.
 
 When a method with this annotation is hit, Spring calls conn.setReadOnly(true), executes the method, and then calls setReadOnly(false) to restore the connection's initial readOnly value. Consequently, every time the method is called, the plugin switches to the reader, executes the method, and then switches back to the writer. Although the reader connection will be cached after the first setReadOnly call, there is still some overhead when switching between the cached writer/reader connections. This constant switching is not an ideal use of the plugin because it is frequently incurring this overhead. The suggested approach for this scenario is to avoid loading the read/write splitting plugin and instead use the writer cluster URL for your write operations and the reader cluster URL for your read operations. By doing this you avoid the overhead of constantly switching between connections while still spreading load across the database instances in your cluster. 
 
 #### Internal connection pools
 
-We recommend that you do not enable internal connection pools when using Spring. This is because Spring by default uses its own external connection pool. The use of both internal and external pools is not tested and may result in problematic behavior.
+We recommend that you do not enable internal connection pools when using Spring with default datasource configuration. This is because Spring by default uses its own external connection pool. The use of both internal and external pools is not tested and may result in problematic behavior. A Spring application should enable internal connection pool only. That's a recommended configuration.
```

Comment:
```suggestion
We recommend that you **disable** internal connection pools when using Spring with default datasource configuration. Spring by default uses its own external connection pool. The use of both internal and external pools has not been tested and may result in problematic behaviour. A Spring application should enable internal connection pool only. That's a recommended configuration.
```

Reply:
I don't understand the last sentence. `A Spring application should enable internal connection pool only. That's a recommended configuration.` but the first sentence said to not use internal connection pool when using Spring with default datasource configuration.

Reply:
I rewrote this part. I hope it's a bit cleaner now.